src/lib/esnext.iterator.d.ts

```diff
@@ -0,0 +1,125 @@
+/// <reference lib="es2015.iterable" />
+export {};
+
+// Abstract type that allows us to mark `next` as `abstract`
+declare abstract class Iterator<T> {
+    abstract next(value?: undefined): IteratorResult<T, void>;
+}
+
+// Merge all members of `BuiltinIterator<T>` into `Iterator<T>`
+interface Iterator<T> extends globalThis.BuiltinIterator<T, void, undefined> {}

```

Comment:
Should we add `TReturn` and `TNext` type parameters to `Iterator`?

We also probably shouldn't specify `TNext` as `undefined` since builtins generally don't actually care what the argument is. You could theoretically have a generator that takes `number | undefined` for `TNext`, but it wouldn't be able to delegate to `yield* [1, 2]` since `number | undefined` isn't assignable to `undefined`.

Reply:
I've opted to leave `TReturn` and `TNext` off of the `Iterator<T>` class for now, though we may want to revisit that after we ship the beta.